src/System.Diagnostics.TraceSource/tests/TestTraceListener.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Diagnostics.TraceSourceTests
+{
+    // mock TraceListener to make assertions against TraceSource behavior
+    internal class TestTraceListener : TraceListener
+    {
+        public enum Method
+        {
+            TraceEvent = 0,
+            TraceData,
+            Dispose,
+            Write,
+            WriteLine,
+        }
+
+        int[] _calls;
+
+        /// <summary>
+        /// Gets the number of times any of the public methods on this instance were called.
+        /// </summary>
+        public int GetCallCount(Method group) {
+            return _calls[(int) group]; 
+        }
+
+        private void Call(Method method)
+        {
+            _calls[(int)method]++;
+        }
+        
+        public TestTraceListener()
+        {
+            _calls = new int[5];
```

Comment:
You could use reflection to get at the number of enum values rather than hardcoding it.  Or if not, at least add a comment indicating that that's what this number is, and maybe a comment on the enum type itself indicating that if a new element is added, this number must be changed.